### PR TITLE
Add option to generate unique component keys for table rows

### PR DESCRIPTION
**Pull request by Argn0**
fixes https://github.com/odota/web/issues/1499

this is to prevent unexpected behavior in all tables where we want to shift around rows. like for the Teamfight table.

### CODE REVIEW

File: src/components/Table/Table.jsx [hunk 1]
```diff
@@ -176,7 +178,7 @@ class Table extends React.Component {
               </MaterialTableHeader>
               <MaterialTableBody displayRowCheckbox={false} selectable={false}>
                 {data.map((row, index) => (
-                  <MaterialTableRow key={index} style={rowStyle(highlightFn, row)}>
+                  <MaterialTableRow key={(uniqueKey && uuid.v4()) || index} style={rowStyle(highlightFn, row)}>
```
Comment (howardchung) [inline src/components/Table/Table.jsx]: perhaps we should use uuid.v4() always?  Index is not a good key for a table row anyway.

Comment (coreymaher) [reply src/components/Table/Table.jsx]: Doesn't this mean every render will generate a new key?
I think we probably want to actually come up with unique keys for each row so the DOM nodes can be reused when possible.

Comment (Argn0) [reply src/components/Table/Table.jsx]: yea it does. I'm already looking for an alternative. I figured it out too late.

**Comment by Argn0** [comment]
Teamfight is the only component where we shift row positions. So this is a reasonable solution I think.

File: src/components/Match/TeamTable.jsx [hunk 2]
```diff
@@ -7,6 +7,8 @@ import { connect } from 'react-redux';
 import Table from 'components/Table';
 import PicksBans from './Overview/PicksBans'; // Displayed only on `Overview` page
 
+const uniqueKey = row => row && row.player_slot + 1;
```
Comment (howardchung) [inline src/components/Match/TeamTable.jsx]: i'd probably call this keyFn for consistency

Comment (howardchung) [reply src/components/Match/TeamTable.jsx]: the prop name, anyway, you could name this the same if you like

**Pull request merged**